File: library/src/main/java/com/getstream/sdk/chat/storage/Sync.java
package com.getstream.sdk.chat.storage;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class Sync {

    public static final int IN_MEMORY = -1;
    public static final int LOCAL_ONLY = 0;
    public static final int LOCAL_UPDATE_PENDING = 1;
    public static final int SYNCED = 2;
    public static final int LOCAL_FAILED = 3;

    public Sync(@Status int status) {
        System.out.println("status :" + status);
    }

    @IntDef({IN_MEMORY, LOCAL_ONLY, LOCAL_UPDATE_PENDING, SYNCED, LOCAL_FAILED})
    @Retention(RetentionPolicy.SOURCE)
    public @interface Status {
    }
}

File: app/src/main/java/io/getstream/chat/android/core/poc/library/Sync.kt
package io.getstream.chat.android.core.poc.library

import androidx.annotation.IntDef


class Sync(@Status status: Int) {
    @IntDef(
        IN_MEMORY,
        LOCAL_ONLY,
        LOCAL_UPDATE_PENDING,
        SYNCED,
        LOCAL_FAILED
    )
    @kotlin.annotation.Retention(AnnotationRetention.SOURCE)
    annotation class Status

    companion object {
        const val IN_MEMORY = -1
        const val LOCAL_ONLY = 0
        const val LOCAL_UPDATE_PENDING = 1
        const val SYNCED = 2
        const val LOCAL_FAILED = 3
    }

    init {
        println("status :$status")
    }
}